debug_toolbar/store.py

```diff
@@ -140,5 +141,91 @@ def panels(cls, request_id: str) -> Any:
             yield panel, deserialize(data)
 
 
+class DatabaseStore(BaseStore):
+    @classmethod
+    def _cleanup_old_entries(cls):
+        """
+        Enforce the cache size limit - keeping only the most recently used entries
+        up to RESULTS_CACHE_SIZE.
+        """
+        # Get the cache size limit from settings
+        cache_size = dt_settings.get_config()["RESULTS_CACHE_SIZE"]
+
+        # Determine which entries to keep (the most recent ones up to cache_size)
+        keep_ids = list(
+            HistoryEntry.objects.all()[:cache_size].values_list("request_id", flat=True)
+        )
+
+        # Delete all entries not in the keep list
+        if keep_ids:
+            HistoryEntry.objects.exclude(request_id__in=keep_ids).delete()
+
+    @classmethod
+    def request_ids(cls):
+        """Return all stored request ids within the cache size limit"""
+        cache_size = dt_settings.get_config()["RESULTS_CACHE_SIZE"]
+        return list(
+            HistoryEntry.objects.all()[:cache_size].values_list("request_id", flat=True)
+        )
+
+    @classmethod
+    def exists(cls, request_id: str) -> bool:
+        """Check if the given request_id exists in the store"""
+        return HistoryEntry.objects.filter(request_id=request_id).exists()
+
+    @classmethod
+    def set(cls, request_id: str):
+        """Set a request_id in the store and clean up old entries"""
+        # Create the entry if it doesn't exist (ignore otherwise)
+        HistoryEntry.objects.get_or_create(request_id=request_id)
+
+        # Enforce the cache size limit to clean up old entries
+        cls._cleanup_old_entries()
+
+    @classmethod
+    def clear(cls):
+        """Remove all requests from the store"""
+        HistoryEntry.objects.all().delete()
+
+    @classmethod
+    def delete(cls, request_id: str):
+        """Delete the stored request for the given request_id"""
+        HistoryEntry.objects.filter(request_id=request_id).delete()
+
+    @classmethod
+    def save_panel(cls, request_id: str, panel_id: str, data: Any = None):
```

Comment:
Similar to `set()`, do we want to use `transaction.atomic()` here as well?

Reply:
I meant this comment for both methods.

Reply:
As per our call, there's a concern about panels updating this concurrently and needing to avoid clearing another panel's data when saving.